perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/TasksManagerBlImpl.java

```diff
@@ -0,0 +1,604 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.RowMapper;
+import java.text.SimpleDateFormat;
+import java.time.format.DateTimeFormatter;
+import java.time.format.ResolverStyle;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * TasksManagerBlImpl
+ */
+public class TasksManagerBlImpl implements TasksManagerBl {
+	@Autowired
+	private ServicesManagerBl servicesManagerBl;
+	@Autowired
+	protected PerunBl perun;
+	@Autowired
+	protected TasksManagerImplApi tasksManagerImpl;
+
+	public TasksManagerBlImpl(TasksManagerImpl tasksManagerImpl) {
+		this.tasksManagerImpl = tasksManagerImpl;
+	}
+
+	@Override
+	public int insertNewTaskResult(TaskResult taskResult, int engineID) throws InternalErrorException {
+		return tasksManagerImpl.insertNewTaskResult(taskResult, engineID);
+	}
+
+	@Override
+	public List<TaskResult> getTaskResults(int engineID) {
+		return getTaskResults(engineID);
+	}
+
+	@Override
+	public TaskResult getTaskResultById(int taskResultId, int engineID) {
+		return tasksManagerImpl.getTaskResultById(taskResultId, engineID);
+	}
+
+	@Override
+	public int clearByTask(int taskId, int engineID) {
+		return tasksManagerImpl.clearByTask(taskId, engineID);
+	}
+
+	@Override
+	public int clearByTask(int taskId) {
+		return tasksManagerImpl.clearByTask(taskId);
+	}
+
+	@Override
+	public int clearAll(int engineID) {
+		return tasksManagerImpl.clearAll(engineID);
+	}
+
+	@Override
+	public int clearAll() {
+		return tasksManagerImpl.clearAll();
+	}
+
+	@Override
+	public int clearOld(int engineID, int numDays) {
+		return tasksManagerImpl.clearOld(engineID, numDays);
+	}
+
+	@Override
+	public List<TaskResult> getTaskResultsByTask(int taskId, int engineID) {
+		return tasksManagerImpl.getTaskResultsByTask(taskId, engineID);
+	}
+
+	@Override
+	public List<TaskResult> getTaskResultsByTaskOnlyNewest(int taskId) {
+		return getTaskResultsByTaskOnlyNewest(taskId);
+	}
+
+	@Override
+	public List<TaskResult> getTaskResultsByTaskAndDestination(int taskId, int destinationId) {
+		return tasksManagerImpl.getTaskResultsByTaskAndDestination(taskId, destinationId);
+	}
+
+	public List<TaskResult> getTaskResultsForDestinations(List<String> destinationsNames) throws InternalErrorException {
+		return tasksManagerImpl.getTaskResultsForDestinations(destinationsNames);
+	}
+
+	/**
+	 * Method create formatter with default settings for perun timestamps and set lenient on false
+	 * Timestamp format:  "dd-MM-yyyy HH:mm:ss" - "ex. 01-01-2014 10:10:10"
+	 *
+	 * Lenient on false means that formatter will be more strict to creating timestamp from string
+	 *
+	 * IMPORTANT: SimpleDateFormat is not thread safe !!!
+	 *
+	 * @return date formatter
+	 */
+	public static SimpleDateFormat getDateFormatter() {
```

Comment:
Whole part between lines ``115`` and ``202`` is unused and was probably just forgotten - its already present and used in ``TasksManagerImpl``.